Tutorial.md

```diff
@@ -0,0 +1,530 @@
+LOGGER.log('UPDATE progress', { ...progress });
+return progress;
+```
+
+## Implement the Delete Handler<a name="resource-type-walkthrough-implement-delete"></a>
+
+Replace the contents of the `delete` handler with:
+
+```
+LOGGER.debug('DELETE request', request);
+const model: ResourceModel = request.desiredResourceState;
+const progress = ProgressEvent.progress<ProgressEvent<ResourceModel, CallbackContext>>();
+const response: Response = await fetch(`${API_ENDPOINT}/${model.uid}`, {
+    method: 'DELETE',
+    headers: DEFAULT_HEADERS,
+});
+await checkedResponse(response, model.uid);
+progress.status = OperationStatus.Success;
+LOGGER.log('DELETE progress', { ...progress });
+return progress;
+```
+
+## Implement the Read Handler<a name="resource-type-walkthrough-implement-read"></a>
+
+Replace the contents of the `read` handler with:
+
+```
+LOGGER.debug('READ request', request);
+const model: ResourceModel = request.desiredResourceState;
+const response: Response = await fetch(`${API_ENDPOINT}/${model.uid}`, {
+    method: 'GET',
+    headers: DEFAULT_HEADERS,
+});
+const jsonData: any = await checkedResponse(response, model.uid);
+model.name = jsonData['name'];
+model.color = jsonData['color'];
+const progress: ProgressEvent<ResourceModel> = ProgressEvent.builder()
+    .status(OperationStatus.Success)
+    .resourceModel(model)
+    .build() as ProgressEvent<ResourceModel>;
+LOGGER.log('READ progress', { ...progress });
+return progress;
+```
+
+## Implement the List Handler<a name="resource-type-walkthrough-implement-list"></a>
+
+Replace the contents of the `list` handler with:
+
+```
+LOGGER.debug('LIST request', request);
+const response: Response = await fetch(API_ENDPOINT, {
+    method: 'GET',
+    headers: DEFAULT_HEADERS,
+});
+const jsonData: any[] = await checkedResponse(response);
+const models: Array<ResourceModel> = jsonData.map((unicorn: any) => {
+    return new ResourceModel({
+        uid: unicorn['_id'],
+        name: unicorn['name'],
+        color: unicorn['color'],
+    });
+});
+const progress: ProgressEvent<ResourceModel> = ProgressEvent.builder()
+    .status(OperationStatus.Success)
+    .resourceModels(models)
+    .build() as ProgressEvent<ResourceModel>;
+LOGGER.log('LIST progress test logger', { ...progress });
+return progress;
+```
+
+After all the handlers have been implemented run `npm build`.
+
+## Create the SAM Test Files and test the handlers<a name="resource-type-walkthrough-test-files"></a>
+
+1. Create five files:
+   + `testing-unicorn-maker/sam-tests/create.json`
+   + `testing-unicorn-maker/sam-tests/read.json`
+   + `testing-unicorn-maker/sam-tests/update.json`
+   + `testing-unicorn-maker/sam-tests/delete.json`
+   + `testing-unicorn-maker/sam-tests/list.json`
+
+2. In `testing-unicorn-maker/sam-tests/create.json`, paste the following test\.
+
+    **Note:**
+    Add the necessary information, such as credential and remove any comments in the file before testing\.
```

Comment:
there is no need to have the user add valid aws credentials. they can just be dummy ones